main/maven/src/mill/main/maven/BuildGen.scala

```diff
@@ -0,0 +1,442 @@
+package mill.main.maven
+
+import mainargs.{Flag, ParserForClass, arg, main}
+import mill.main.build.{BuildCompanion, BuildDefinition, Node}
+import mill.runner.FileImportGraph.backtickWrap
+import org.apache.maven.model.{Dependency, Model}
+
+import scala.collection.immutable.{SortedMap, SortedSet}
+import scala.jdk.CollectionConverters.*
+
+/**
+ * Converts a Maven build to Mill by generating Mill build file(s) from POM file(s).
+ *
+ * The generated output should be considered scaffolding and will likely require edits to complete conversion.
+ *
+ * ===Features===
+ *  - converts nested modules
+ *  - captures publish metadata
+ *  - configures dependencies for scopes:
+ *    - compile
+ *    - provided
+ *    - runtime
+ *    - test
+ *  - configures testing frameworks:
+ *    - JUnit 4
+ *    - JUnit 5
+ *    - TestNG
+ *  - supports Maven plugins:
+ *    - org.apache.maven.plugins:maven-compiler-plugin
+ *    - org.codehaus.mojo:build-helper-maven-plugin
+ *      - path filters are not supported
+ *
+ * ===Limitations===
+ *  - build extensions are not supported
+ *  - build profiles are not supported
+ *  - packaging support is limited to
+ *    - jar
+ *    - pom
+ */
+@mill.api.internal
+object BuildGen {
+
+  def main(args: Array[String]): Unit = {
+    implicit val cfg: BuildGenConfig = ParserForClass[BuildGenConfig].constructOrExit(args.toSeq)
```

Comment:
Let's make `cfg: BuildGenConfig` an explicit parameter and just pass it everywhere manually; it's a small enough part of our codebase that passing it explicitly isn't too much of a burden